### PR TITLE
Adding mask-rcnn sagemaker model parallel example notebook. 

### PR DESCRIPTION
**Pull request by YangFei1990**
*Issue #, if available:*

*Description of changes:*
Adding a notebook tutorial using **sagemaker distributed model parallel** with mask-rcnn model. This notebook also supports to create a custom VPC for training with S3/FSx/EFS as data input channels, which are adopted from the [distributed_tensorflow_mask_rcnn](https://github.com/aws/amazon-sagemaker-examples/tree/master/advanced_functionality/distributed_tensorflow_mask_rcnn).
Since there are code blocks that must be filled by users interactively, e.g. FSx file system id:
```
file_system_id = # Spercify your file system id
```
This tutorial will fail to run the end-to-end test from CI. I have tested this tutorial end-to-end, from creating the VPC with `stack-sm.sh` to using all three data input methods for training. I was running the notebook with `conda_python3` kernel.

When reviewing, the files inside `mask-rcnn/utils/train_scripts` could be ignored or briefly reviewed, since those are source code for mask-rcnn model. But those source code scripts should be included since they will make it easy for user to modify the model and run the training again.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by seanpmorgan** [comment]
eek 250 files? Anyway we can leverage maskrcnn benchmark repo instead?
https://github.com/facebookresearch/maskrcnn-benchmark

**Comment by YangFei1990** [comment]
> eek 250 files? Can we leverage maskrcnn benchmark repo instead?
> https://github.com/facebookresearch/maskrcnn-benchmark

The model code is from https://github.com/karakusc/training_results_v0.7/tree/smp-mrcnn. I have already simplified the repo and only kept necessary files. The reason to have the model code in this PR is to provide the support to customers so that they can view and modify the model code while using this tutorial, since model like mask-rcnn has many variations(different heads, back-bones, etc..). 
Also since this mrcnn model requires custom ops, which takes a very long time to build, the current notebook allows the user to first build the custom ops, then copy the model code into the container, which makes it much easier for the user to run with some modification of the model code.
Maybe I can further simplify the model folder but it will still contains many files. Is that acceptable?

**Comment by hsl89** [comment]
pending reason: requires user input for Fxs id

File: training/distributed_training/pytorch/model_parallel/mask-rcnn/smp_maskrcnn_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,649 @@
+{
```
Comment (hsl89) [inline training/distributed_training/pytorch/model_parallel/mask-rcnn/smp_maskrcnn_tutorial.ipynb:1]: <p>set default mode </p><pre class="ql-syntax" spellcheck="false">mode = None # replace None with your file system 
if mode == None:
  mode = "S3"
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1965/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (hsl89) [inline training/distributed_training/pytorch/model_parallel/mask-rcnn/smp_maskrcnn_tutorial.ipynb:1]: <p>Get subnet id and security group id via boto3 </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1965/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (hsl89) [inline training/distributed_training/pytorch/model_parallel/mask-rcnn/smp_maskrcnn_tutorial.ipynb:1]: <p>Get <code>file_system_id</code>  by <code>boto3</code>  or whatever automated way</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1965/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (hsl89) [inline training/distributed_training/pytorch/model_parallel/mask-rcnn/smp_maskrcnn_tutorial.ipynb:1]: <p>Use <code>hyperparameters</code>  to control what percent of data to train and how many epochs to train. Default to small numbers so that it can be tested </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1965/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>
